### PR TITLE
Qutrit channel amplitude damping

### PR DESCRIPTION
**Pull request by Gabriel-Bottrill**
**Context:**
`default.qutrit.mixed` device has been added, but only one channel has been added, [qml.QutritDepolarizingChannel](https://github.com/PennyLaneAI/pennylane/pull/5502/), this adds the second channel to the device so the device can simulate the qutrit equivalent of amplitude damping noise.

**Description of the Change:**
Adds new channel module to `qml.ops.qutrit` package. Adding the second qutrit channel `QutritAmplitudeDamping`.

**Benefits:**
Allows for `defualt.qutrit.mixed` to simulate amplitude damping noise. Making `default.qutrit.mixed`, much more useful.

**Possible Drawbacks:**
N/A

**Related GitHub Issues:**
N/A

### CODE REVIEW

File: tests/ops/qutrit/test_qutrit_channel_ops.py
Comment (Gabriel-Bottrill) [inline tests/ops/qutrit/test_qutrit_channel_ops.py:78]: Added this to ensure test is using the expected method of grad computation.

File: tests/ops/qutrit/test_qutrit_channel_ops.py
Comment (Gabriel-Bottrill) [inline tests/ops/qutrit/test_qutrit_channel_ops.py:116]: I imported math from pennylane, I ended up removing all calls to `qml.math` and replaced with `math`

File: tests/ops/qutrit/test_qutrit_channel_ops.py
Comment (Gabriel-Bottrill) [inline tests/ops/qutrit/test_qutrit_channel_ops.py:209]: I calculated these partials by hand, I can add notes, but they are fairly simple partial derivatives.

File: pennylane/ops/qutrit/channel.py
Comment (Gabriel-Bottrill) [inline pennylane/ops/qutrit/channel.py:303]: gamma_1 and gamma_2 are checked separately if they are abstract, I believe this would be the correct behaviour, but if  I am mistaken and both should be abstract if one is then please let me know and I will fix it.

**Review by Gabriel-Bottrill — COMMENTED** [review]
@albi3ro, @glassnotes, @mudit2812, and @trbromley I have finished this PR and added some comments. I look forward to your reviews!

File: pennylane/ops/qutrit/channel.py
Comment (glassnotes) [inline pennylane/ops/qutrit/channel.py:231]:
```suggestion
    This can be modelled by the qutrit amplitude damping channel with the following Kraus matrices:
```

File: pennylane/ops/qutrit/channel.py
Comment (glassnotes) [inline pennylane/ops/qutrit/channel.py:251]:
```suggestion
                \end{bmatrix},

    where :math:`\gamma_1 \in [0, 1]` and :math:`\gamma_2 \in [0, 1]` are the amplitude damping
    probabilities for subspaces (0,1) and (0,2) respectively.
```

Comment (glassnotes) [reply pennylane/ops/qutrit/channel.py:251]: Instead of "probabilities for subspaces" it may be clearer to phrase this in terms of transitions from 2->0 and 1->0, since it is not a bi-directional operation (in the sense that we don't have 0->2 as we would for a normal subspace rotation)

File: pennylane/ops/qutrit/channel.py
Comment (glassnotes) [inline pennylane/ops/qutrit/channel.py:271]: No analytic diff possible here? Or just not worked out yet?

File: pennylane/ops/qutrit/channel.py
Comment (glassnotes) [reply pennylane/ops/qutrit/channel.py:303]: Good question - presumably both would be passed in the same interface, but I think it doesn't hurt to check both. However the for loop around the if statement has redundancy, either check one at a time or use the iterating variable from the loop. May also be good to include the explicit ID of any offending gammas so the user knows if it's 1 or 2.

File: pennylane/ops/qutrit/channel.py
Comment (glassnotes) [inline pennylane/ops/qutrit/channel.py:304]: This is gonna sound silly but, we can write if statements like this in python now? I thought the conditions had to be checked separately

File: tests/ops/qutrit/test_qutrit_channel_ops.py
Comment (glassnotes) [reply tests/ops/qutrit/test_qutrit_channel_ops.py:78]: See point above about A vs. F; I thought param shift is an A method?

File: tests/ops/qutrit/test_qutrit_channel_ops.py
Comment (glassnotes) [inline tests/ops/qutrit/test_qutrit_channel_ops.py:178]:
```suggestion
        """Test gamma_1=gamma_2=0 gives correct Kraus matrices"""
```

File: tests/ops/qutrit/test_qutrit_channel_ops.py
Comment (glassnotes) [inline tests/ops/qutrit/test_qutrit_channel_ops.py:186]:
```suggestion
        """Test the correct Kraus matrices are returned, also ensures that the sum of gammas can be over 1."""
```

Comment (glassnotes) [reply tests/ops/qutrit/test_qutrit_channel_ops.py:186]: This second condition is checked only implicitly; is it necessary to mention?

File: tests/ops/qutrit/test_qutrit_channel_ops.py
Comment (glassnotes) [inline tests/ops/qutrit/test_qutrit_channel_ops.py:174]: Bigger picture question: neither this PR nor #5502 included integration tests that involve actually using these channels in QNodes. The AmpDamp ones should be added here, but can you add some TODOs here please and include the Depolarizing ones in a follow-up PR?

**Review by glassnotes — CHANGES_REQUESTED** [review]
@Gabriel-Bottrill looks great! Small comment re. gradient, and realized this and previous PR are missing integration tests for the channels.

File: doc/releases/changelog-dev.md
Comment (trbromley) [inline doc/releases/changelog-dev.md:106]:
```suggestion
* `qml.QutritAmplitudeDamping` channel has been added, allowing for noise processes modelled by amplitude damping to be simulated on the `default.qutrit.mixed` device.
```
Minor point but the single backticks should be fine for this markdown file.

File: pennylane/ops/qutrit/channel.py
Comment (trbromley) [inline pennylane/ops/qutrit/channel.py:229]: I wonder why it shows up like this in the docs:
image

We should be able to manually override that though, e.g., we [do it here](https://github.com/PennyLaneAI/pennylane/blob/59c32874dc825af65a86026ea51f68b9338446a4/pennylane/ops/qubit/non_parametric_ops.py#L35).

Comment (trbromley) [inline pennylane/ops/qutrit/channel.py:229]:
```suggestion
    Single-qutrit amplitude damping error channel.

```

**Review by trbromley — COMMENTED** [review]
Thanks a lot @Gabriel-Bottrill!

File: pennylane/ops/qutrit/channel.py
Comment (mudit2812) [inline pennylane/ops/qutrit/channel.py:278]:
```suggestion
        r"""Kraus matrices representing the ``QutritAmplitudeDamping`` channel.
```

File: pennylane/ops/qutrit/channel.py
Comment (mudit2812) [reply pennylane/ops/qutrit/channel.py:304]: Yep, this is valid :)

File: pennylane/ops/qutrit/channel.py
Comment (mudit2812) [inline pennylane/ops/qutrit/channel.py:303]: I think it makes more sense to validate this in `__init__` rather than here. Right now, if a user creates a `QutritAmplitudeDamping` operator, it will only raise an error when the matrix is created.

File: tests/ops/qutrit/test_qutrit_channel_ops.py
Comment (mudit2812) [reply tests/ops/qutrit/test_qutrit_channel_ops.py:174]: I thought about this a lot, and my personal opinion is that we've validated that arbitrary channels are applied correctly to the state vector in `tests/devices/qutrit_mixed/test_qutrit_mixed_apply_operations.py`, so we don't need to validate that specific channels are applied correctly. We would be bloating the test suite unnecessarily. What I do recommend adding is integration/end-to-end testing in `tests/devices/test_default_qutrit_mixed.py` that validates larger workflows, such as qnode differentiation.

**Review by mudit2812 — COMMENTED** [review]
Looks great! Happy to approve after the open comments are addressed.

File: pennylane/ops/qutrit/channel.py
Comment (Gabriel-Bottrill) [reply pennylane/ops/qutrit/channel.py:272]: Not worked out, it isn't worked out for the qubit case either. I will try to find an analytic differentiation method.

File: tests/ops/qutrit/test_qutrit_channel_ops.py
Comment (Gabriel-Bottrill) [reply tests/ops/qutrit/test_qutrit_channel_ops.py:78]: This is a fix to the previous PR [QutritDepolarizingChannel](https://github.com/PennyLaneAI/pennylane/pull/5502), I should have implemented it there, however the function of that PR is unaffected, this ensures the test is testing the correct measurement.

File: tests/ops/qutrit/test_qutrit_channel_ops.py
Comment (Gabriel-Bottrill) [reply tests/ops/qutrit/test_qutrit_channel_ops.py:174]: I agree, @mudit2812  would you think the larger integration testing should be in this PR or a separate PR?

File: pennylane/ops/qutrit/channel.py
Comment (Gabriel-Bottrill) [reply pennylane/ops/qutrit/channel.py:304]: This is similar to other operators in PennyLane, including the qubit `AmplitudeDamping` channel. I thought it was due to the potential for the parameter value to be changed without making a new operator?

**Review by Gabriel-Bottrill — COMMENTED** [review]
I comment on a few open questions, there are still a few open questions that I will address shortly.

File: pennylane/ops/qutrit/channel.py
Comment (Gabriel-Bottrill) [reply pennylane/ops/qutrit/channel.py:229]: @trbromley  I can do that, but it seems like there may be a bigger issue. I just checked other operators and they all seem to be doing this. For example this behavior is happening in [PR 5676 for the RX gate](https://xanaduai-pennylane--5668.com.readthedocs.build/en/5668/code/api/pennylane.RX.html). This is different than the stable version. I tracked back the first commit who's PRs docs has this occur and I found that [PR 5511](https://xanaduai-pennylane--5511.com.readthedocs.build/en/5511/code/api/pennylane.RX.html) seems to be the start. I am just using the docs built by CI so it's likely I'm wrong about which PR is the first, but I think that there is a bug here. I'd be happy to do some more exploration and build the docs myself for each commit to make a bug report if that would help.

Comment (Gabriel-Bottrill) [reply pennylane/ops/qutrit/channel.py:229]: Update: I tested the docs on my machine for the commit where #5511 is added and found that it had the same behavior as in the CI generated one.
I also tested the docs locally for the commit for #5595 the PR that was merged directly before #5511. In this commit the docs had the expected values in the arguments for functions, ie they were generated properly.

Note: This only seems to be happening for docs of Operator class types. This is consistent with what was changed in 5511.

File: pennylane/ops/qutrit/channel.py
Comment (Gabriel-Bottrill) [reply pennylane/ops/qutrit/channel.py:272]: I just did the math, I'm fairly certain there is no param-shift possible here.

File: pennylane/ops/qutrit/channel.py
Comment (glassnotes) [reply pennylane/ops/qutrit/channel.py:229]: I think this is out of scope for this PR, if there is a deeper underlying problem, we can make an issue for it and then reference it in a comment in this discussion thread so the source can be easily found.

File: pennylane/ops/qutrit/channel.py
Comment (glassnotes) [reply pennylane/ops/qutrit/channel.py:272]: Okay, thanks for checking! Does this necessitate using F instead of A then?

File: tests/ops/qutrit/test_qutrit_channel_ops.py
Comment (glassnotes) [reply tests/ops/qutrit/test_qutrit_channel_ops.py:174]: Thanks for your input on this @mudit2812 !  I suggest putting this in a separate PR since it is something that would presumably cover more than this channel.

**Review by glassnotes — APPROVED** [review]
This looks good to me - thanks @Gabriel-Bottrill ! I will keep an eye on the remaining discussions, but I leave it to Mudit or Tom to give the final approval here.

File: pennylane/ops/qutrit/channel.py
Comment (Gabriel-Bottrill) [reply pennylane/ops/qutrit/channel.py:272]: I believe so since if it is left as none then it would mean that it could not be differentiated, it can be there is just no nice method for doing so.

File: tests/ops/qutrit/test_qutrit_channel_ops.py
Comment (Gabriel-Bottrill) [reply tests/ops/qutrit/test_qutrit_channel_ops.py:174]: Ok, I will make a separate PR, I will close this conversation now though.

File: pennylane/ops/qutrit/channel.py
Comment (Gabriel-Bottrill) [reply pennylane/ops/qutrit/channel.py:229]: I have made a bug report #5724.

File: pennylane/ops/qutrit/channel.py
Comment (mudit2812) [reply pennylane/ops/qutrit/channel.py:304]: We're generally against mutating operators in-place. I would still say that such validation should be happening in the `__init__` method. Seems to me like `AmplitudeDamping` should be changed as well (not a task for you though)

Comment (Gabriel-Bottrill) [reply pennylane/ops/qutrit/channel.py]: Ok I have updated the verification

**Review by mudit2812 — APPROVED** [review]

**Comment by mudit2812** [comment]
The jax failures should fix themselves once you update the branch @Gabriel-Bottrill

**Comment by Gabriel-Bottrill** [comment]
Thank you @mudit2812!

**Pull request merged**
